### PR TITLE
4710 allow download of sequences by seqids

### PR DESCRIPTION
**Pull request by mannSheth**
Download fastq or fasta through listing samplesIds

Usages:
image

image

image
skips incorrect items

Added extended permissions so that a seqViewer can also fetch sequences 
image
However only for the ones it has access to.

### CODE REVIEW

File: austrakka/utils/options.py [hunk 1]
```diff
@@ -202,7 +233,7 @@ def opt_organisation(**attrs: t.Any):
 
 def opt_group(**attrs: t.Any):
     defaults = {
-        'required': True,
+        'required': False,
```
Comment (claresloggett) [inline austrakka/utils/options.py:236]: I think only the sequence commands are using this option, but given they're all passing in required=True, what's the motivation for changing the default to False?

I've also just noticed we have both opt_group and opt_group_name representing the same --group-name parameter, this was a preexisting problem by the looks of it! These should be consolidated. opt_group is currently multiple, but it doesn't need to be, as we ALSO have opt_groups which is multiple.

File: austrakka/components/sequence/funcs.py [hunk 2]
```diff
@@ -395,14 +408,34 @@ def _get_seq_api(group_name: str):
     return api_path
 
 
+def _get_seq_api_sample_names(sample_ids: List[str]):
+    api_path = SEQUENCE_PATH
```
Comment (claresloggett) [inline austrakka/components/sequence/funcs.py:412]: I think using this function will require a server-side change to make it work for non-admins. The CLI command needs to be tested using a non-admin account.

**Review by claresloggett — CHANGES_REQUESTED** [review]

File: austrakka/utils/options.py (hunk 1 above)
Comment (mannSheth) [reply austrakka/utils/options.py:236]: I think I will go through this with you so I don't make a mistake when culling duplicates and changing misused options

Comment (mannSheth) [reply austrakka/utils/options.py]: I think I resolved the duplicate issue with sam 👍

File: austrakka/components/sequence/funcs.py (hunk 2 above)
Comment (mannSheth) [reply austrakka/components/sequence/funcs.py:440]: It should be working now. I did test it and added the permission to this branch: [4718](https://github.com/AusTrakka/austrakka-portal/pull/1213)

**Review by claresloggett — APPROVED** [review]

**Pull request merged**